.github/workflows/proto.yml

```diff
@@ -1,13 +1,30 @@
-name: buf
+name: Proto
 on:
   pull_request:
   push:
     branches:
         - main
 jobs:
-  build-lint-push-protos:
+  changes:
     runs-on: ubuntu-latest
-    environment: BUF
+    permissions:
+      pull-requests: read
+    outputs:
+      proto: ${{ steps.filters.outputs.proto }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
+        id: filters
+        with:
+          filters: |
+            proto:
+              - 'crates/astria-proto/**'
+              - '.github/workflows/proto.yml'
+  
+  build-lint:
```

Comment:
Nit: `lint-protos` might be more descriptive.

Reply:
Technically right now all it does is build and makes sure it completes. The lint is actually commented out since we are still in the "it's all alpha break it fine" stage.